src/odemis/driver/zeiss.py

```diff
@@ -0,0 +1,963 @@
+        except Exception:
+            logging.exception("Unexpected failure when updating position")
+
+    def _doMoveRel(self, future, shift):
+        """
+        move by the shift
+        shift (dict with "z" key): unit m
+        """
+        z = self.parent.GetFocus()
+        z += round(int(shift["z"] * 1e3))
+
+        with future._moving_lock:
+            try:
+                if future._must_stop.is_set():
+                    future._was_stopped = True
+                    raise CancelledError()
+                else:
+                    self.parent.SetFocus(z)
+
+            except RemconError:
+                if future._must_stop.is_set():
+                    future._was_stopped = True
+                    raise CancelledError()
+                else:
+                    raise
+            finally:
+                # Update the position, even if the move didn't entirely succeed
+                self._updatePosition()
+
+    def _doMoveAbs(self, future, pos):
+        """
+        move to pos
+        pos (dict with "z" key): unit m
+        """
+
+        with future._moving_lock:
+            try:
+                if future._must_stop.is_set():
+                    future._was_stopped = True
+                    raise CancelledError()
+
+                # TODO: there is a possible race condition if the cancel happens
+                # just now, before the move request is sent. To fix it, we'd need
+                # to separate the move request from waiting for the answer.
+                # It's currently not an issue anyway as we need a work-around for
+                # the SU5000 not stopping the move if the stop request is too
+                # soon after the move request.
+                else:
+                    self.parent.SetFocus(pos["z"])
+
+            except RemconError:
+                if future._must_stop.is_set():
+                    future._was_stopped = True
+                    raise CancelledError()
+                else:
+                    raise
+            finally:
+                # Update the position, even if the move didn't entirely succeed
+                self._updatePosition()
+
+    def _cancelCurrentMove(self, future):
+        """
+        Cancels the current move (both absolute or relative). Non-blocking.
+        future (Future): the future to stop. Unused, only one future must be
+         running at a time.
+        return (bool): True if it successfully cancelled (stopped) the move.
+        """
+        # The difficulty is to synchronise correctly when:
+        #  * the task is just starting (not finished requesting axes to move)
+        #  * the task is finishing (about to say that it finished successfully)
+        logging.debug("Cancelling current move")
+
+        future._must_stop.set()  # tell the thread taking care of the move it's over
+        self.parent.Abort()
+
+        # Note: the SU5000 (software v2.03), doesn't actually stop a move if the
```

Comment:
All these lines don't make sense. It's a work around for the SU5000. => just remove.